### DISCUSSION TITLE
Override GPIO Pins for Switches

### DISCUSSION THREAD

**Discussion by plazarch** [nebhead/PiFire#107] [Q&A]
I am in the middle of my build.  I received my circuit boards yesterday.  Using nebheads board v2.00a and weberboxs switch board v1.00a only the up button was working.  After much troubleshooting, I believe GPIO's 20 and 21 on my Raspberry Pi Zero 2 W might be defective.  I moved the down button to GPIO 13, the enter button to GPIO 12, edited display_ili9341b.py to change the pin numbering, and restarted the server.  Now the switches are working properly.  My question is, is there a place I can put those settings to override the defaults so they won't be overwritten during updates?  Thanks in advance for any assistance, and a HUGE thanks for this project.

**Comment by nebhead** [comment]
You know, you bring up a very good point.  This is going to be a problem for anyone who has a different pin config on their raspberry pi and would be overwritten on updates.  This is something that we should have as a configuration option in the wizard.  And would probably help eliminate the need for some additional files that we have in our build.  

So unfortunately, the short answer is that we don't have a solution for this right now.  However, I think we should create a pin dictionary that is stored in settings and can be edited by the Wizard... or at least manually so that updates to force people to have to edit their display/IO modules manually.  I'll commit to looking into this one - no promises on timing, but I will make sure we can solve this one.  

Thanks for bringing it up!  Hope you enjoy the project!

**Reply by plazarch** [reply]
Thanks for the quick response and I understand about it taking a while.  Would you like me to open a feature request?

**Reply by plazarch** [reply]
Too late.  I see you already did it. :-)

**Reply by nebhead** [reply]
> Too late. I see you already did it. :-)

Yep!  I'm on it. :)
